executor/src/witgen/data_structures/mutable_state.rs

```diff
@@ -0,0 +1,86 @@
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, HashMap},
+};
+
+use powdr_number::FieldElement;
+
+use crate::witgen::{
+    machines::{KnownMachine, Machine},
+    rows::RowPair,
+    EvalError, EvalResult, QueryCallback,
+};
+
+/// The container and access method for machines and the query callback.
+/// The machines contain the actual data tables.
+/// This struct uses interior mutability for accessing the machines.
+pub struct MutableState<'a, 'b, T: FieldElement, Q: QueryCallback<T>> {
+    machines: Vec<RefCell<KnownMachine<'a, T>>>,
+    identity_to_machine_index: BTreeMap<u64, usize>,
+    query_callback: &'b Q,
```

Comment:
Couldn't this be `'a` as well?

Reply:
Probably yes, but it's more annoying lifetime changes because it will lead to Processor requiring fewer lifetimes etc. Will do it in another pr.